Documentation/ApiOverview/CommandControllers/Tutorial.rst

```diff
@@ -0,0 +1,226 @@
+.. include:: /Includes.rst.txt
+
+.. _console-command-tutorial:
+
+========
+Tutorial
+========
+
+Create a console command from scratch
+=====================================
+
+A console command is always situated in an Extension. If you want to create
+one, :ref:`kickstart a custom extension <extension-kickstart>` or use your
+sitepackage extension.
+
+Creating a basic command
+========================
+
+The extension kickstarter ":ref:`Make <extension-make>`" offers a convenient
+console command that creates a new command in an extension of your choice:
+:ref:`Create a new console command with "Make" <extension-make-console-command>`.
+You can use "Make" to create a console command even if your extension was
+created by different means.
+
+This command can be found in the
+`Examples extension <https://github.com/TYPO3-Documentation/t3docs-examples>`.
+
+1. Register the command
+-----------------------
+
+Register via DI in :file:`Configuration/Services.yaml` by adding the service
+definition for your class as tag :yaml:`console.command`:
+
+..  code-block:: yaml
+    :caption: EXT:some_extension/Configuration/Services.yaml
+    :linenos:
+    :emphasize-lines: 12-16
+
+    services:
+      _defaults:
+        autowire: true
+        autoconfigure: true
+        public: false
+
+      T3docs\Examples\:
+        resource: '../Classes/*'
+        exclude: '../Classes/Domain/Model/*'
+
+      T3docs\Examples\Command\DoSomethingCommand:
+        tags:
+          - name: console.command
+            command: 'examples:dosomething'
+            description: 'A command that does nothing and always succeeds.'
+            schedulable: true
+
+..  note::
+    Despite using :file:`autoconfigure: true` the commands
+    have to be explicitly defined in :file:`Configuration/Services.yaml`. The
+    description is mandatory.
+
+2. Create the command class
+---------------------------
+
+Create a class called :php:`DoThingsCommand` extending
+:php:`\Symfony\Component\Console\Command\Command`.
+
+..  include:: /CodeSnippets/Tutorials/Command/Classes/DoSomethingCommand.rst.txt
+
+The following two methods should be overridden by your class:
+
+:php:`configure()`
+    As the name would suggest, allows to configure the command.
+    The method allows to add a help text and/or define arguments.
+
+:php:`execute()`
+    Contains the logic when executing the command. Should return
+    :php:`Command::SUCCESS` or :php:`Command::FAILURE`.
```

Comment:
```suggestion
    an integer: `0` means the command is successful, greater `0` means an error occurred. 
    The :php:`Command class` provides three constants for convenience:
    :php:`Command::SUCCESS`, :php:`Command::FAILURE` and :php:`Command::INVALID`.
```

Reply:
It is best practice to return the constant, not an integer literal

Reply:
Yup, but to know the background is always good. But it is a suggestion, feel free to discard it.

Reply:
I made a suggestion